src/_blog/webtransport.md

```diff
@@ -0,0 +1,247 @@
+
+WebTransport meets their goals and makes it possible to send data quickly on the web.
+
+## The old solution (WebSocket) and its challenges
+
+Before we delve deeper, let's reflect on the history of browser connectivity in libp2p. Questions that come to mind are:
+How did libp2p browser nodes connect to server nodes before the advent of WebTransport, and what challenges existed when bridging browsers to the libp2p ecosystem?
+
+Browsers dial [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) connections (for [HTTP 1.1](https://www.w3.org/Protocols/rfc2616/rfc2616.html) and [HTTP/2](https://web.dev/performance-http2/)) and [QUIC](https://www.rfc-editor.org/rfc/rfc9000.html) connections (for [HTTP/3](https://www.cloudflare.com/learning/performance/what-is-http3/)) all the time. However, there’s no way to *just* dial a TCP or QUIC connection, and use it for other things than HTTP.
+<!-- Explain why -->
+
+This posed a problem when integrating browser applications with libp2p. libp2p is built on top a bidirectional, asynchronous stream abstraction whereas HTTP is a stateless, unidirectional, synchronous request-response protocol. This means in HTTP a client sends a request to a server and waits for it to respond. No state is stored, and the HTTP API only cares about the immediate request. In libp2p, interactions are event based and asynchronous. Communication happens bidirectionally with no need to wait for responses; state is kept around to keep contexts of events that occur during communication.
+
+Therefore, for the longest time, the only way to connect a libp2p client running in the browser to the rest of the network was using the [somewhat dated WebSocket protocol](https://www.rfc-editor.org/rfc/rfc6455).
+
+WebSocket works by first establishing a HTTP(S) connection (most commonly HTTP 1.1) to the server, and then performing a single HTTP request to “upgrade” the connection to a bidirectional byte-stream. In libp2p, this allowed us to then treat this WebSocket connection as a raw TCP connection.
+We run an additional libp2p handshake on that connection and apply a stream multiplexer ([yamux](https://github.com/libp2p/go-yamux) or [mplex](https://github.com/libp2p/specs/tree/master/mplex)). The added step is necessary because WebSocket is susceptible to a problem called [head-of-line blocking](https://hpbn.co/websocket/#websocket-multiplexing-and-head-of-line-blocking). Because the protocol doesn't support multiplexing natively and this can cause one message sent via WebSocket to block other messages behind it (even if the messages are independent of each other.)
+
+This has multiple drawbacks:
+
+- Slow time to connect
+    - It takes 6 network roundtrips until the libp2p connection is finally established because of the steps involved to establish a WebSocket connection.
```

Comment:
```suggestion
    - It takes six network roundtrips until the libp2p connection is finally established because of the steps involved in establishing a WebSocket connection.
```